pyaedt/desktop.py

```diff
@@ -152,69 +192,35 @@ def release_desktop(close_projects=True, close_desktop=True):
         ``True`` when successful, ``False`` when failed.
 
     """
+
     Module = sys.modules['__main__']
-    if sys.modules['__main__'].interpreter == "ironpython":
-        desktop = Module.oDesktop
-        scopeID = desktop.ScopeID
-        i = 0
-        if close_projects:
-            proj_list = desktop.GetProjectList()
-            for prj in proj_list:
-                desktop.CloseProject(prj)
-        if "COMUtil" in dir(Module):
-            while i <= scopeID:
-                Module.COMUtil.ReleaseCOMObjectScope(Module.COMUtil.PInvokeProxyAPI, i)
-                i += 1
-            try:
-                del Module.oDesktop
-                return True
-            except:
-                Module.oMessenger.add_info_message("Attributes not present")
-                return False
-        else:
-            pid = Module.oDesktop.GetProcessID()
-            try:
-                os.kill(pid, 9)
-                return True
-            except:
-                Module.oMessenger.add_error_message("something went wrong in Closing AEDT")
-                return False
+    if "oDesktop" not in dir(Module):
+        _delete_objects()
+        return False
     else:
         desktop = Module.oDesktop
-        i = 0
         if close_projects:
             proj_list = desktop.GetProjectList()
             for prj in proj_list:
                 desktop.CloseProject(prj)
-
-        if close_desktop:
-            scopeID = 5
-            while i <= scopeID:
-                Module.COMUtil.ReleaseCOMObjectScope(Module.COMUtil.PInvokeProxyAPI, 0)
-                i += 1
-            Module = sys.modules['__main__']
-            pid = Module.oDesktop.GetProcessID()
-            try:
-                os.kill(pid, 9)
-                del Module.oDesktop
-                return True
-            except:
-                Module.oMessenger.add_error_message("something went wrong in Closing AEDT")
-                return False
-        else:
+        pid = Module.oDesktop.GetProcessID()
+        if not (is_ironpython and inside_desktop):
+            i = 0
             scopeID = 5
             while i <= scopeID:
                 Module.COMUtil.ReleaseCOMObjectScope(Module.COMUtil.PInvokeProxyAPI,i)
                 i += 1
+            _delete_objects()
+
+        if close_desktop:
             try:
-                del Module.oDesktop
-            except:
-                Module.oMessenger.add_info_message("Attributes not present")
-            try:
-                del Module.pyaedt_initialized
+                os.kill(pid, 9)
+                _delete_objects()
                 return True
             except:
+                warnings.warn("something went wrong in Closing AEDT")
                 return False
+    return True
```

Comment:
```suggestion
        return True
```
I think it is missing a TAB here. Otherwise this `return` will never be reached.

Reply:
It will return if none of previous conditions is reached